pkg/model/awsmodel/api_loadbalancer.go

```diff
@@ -266,6 +272,40 @@ func (b *APILoadBalancerBuilder) Build(c *fi.ModelBuilderContext) error {
 		}
 	}
 
+	// Allow Kops-controller from nodes to ELB, thence to the master instances.
+	if b.UseKopsControllerForNodeBootstrap() && b.Cluster.Spec.API != nil && b.Cluster.Spec.API.LoadBalancer != nil && b.Cluster.Spec.API.LoadBalancer.UseForInternalApi {
+		nodeGroups, err := b.GetSecurityGroups(kops.InstanceGroupRoleNode)
+		if err != nil {
+			return err
+		}
+
+		for _, nodeGroup := range nodeGroups {
+			suffix := nodeGroup.Suffix
+			c.AddTask(&awstasks.SecurityGroupRule{
+				Name:          fi.String(fmt.Sprintf("kops-controller-node%s-to-elb", suffix)),
+				Lifecycle:     b.SecurityLifecycle,
+				FromPort:      fi.Int64(wellknownports.KopsControllerPort),
+				Protocol:      fi.String("tcp"),
+				SecurityGroup: lbSG,
+				SourceGroup:   nodeGroup.Task,
```

Comment:
Can we check if the API server load balancer is public, and if so, use a more lax source range here? As @hakman realized, for a public ELB, the traffic comes in from the nodes via a NAT gateway with a source address that's not even within any of the VPC's CIDR blocks. (The NAT gateways each have a private IP address within the VPC, but the corresponding "Elastic IP address" is outside the VPC.)

Reply:
Per https://github.com/kubernetes/kops/issues/10139#issuecomment-719978826, I now think we're fixing the wrong problem here.

Reply:
If someone asks for Public LB and UseForInternalApi, I think should allow access from 0.0.0.0/0, same as for API.
Or maybe not allow UseForInternalApi in this case, as it won't work anyway.
I don't have a strong preference in general here, so feel free to ignore my comment.

Reply:
I want to make sure I understand your second proposal: Could _kops_ reject "useForInternalApi" as invalid when the load balancer is public? I think that's the best choice, if it's feasible.

Reply:
I liked the option—clearly, as I tried to enable it—for accessing the Kubernetes API servers, not really thinking about whether the load balancer was public or not. At that point, though, I didn't realize that it would be used for other things like this node bootstrapping.

And did this other use start in _kops_ 1.19? I didn't experience this problem with _kops_ 1.18.2.